tests/epics/test_pvi.py

```diff
@@ -144,3 +144,51 @@ async def test_device_create_children_from_annotations(
     assert device.device is block_2_device
     assert device.device.device is block_1_device
     assert device.signal_device is top_block_1_device
+
+
+@pytest.fixture
+def pvi_test_device_with_device_vectors_t():
+    """A fixture since pytest discourages init in test case classes"""
+
+    class TestDevice(Block2):
+        def __init__(self, prefix: str, name: str = ""):
+            self._prefix = prefix
+            create_children_from_annotations(self, device_vectors={"device_vector": 2})
```

Comment:
Add an `optional_signal_rw: Optional[SignalRW[int]]` to `Block2` and check
```suggestion
            create_children_from_annotations(
                self,
                included_optional_fields=("optional_signal_rw",),
                device_vectors={"device_vector": 2}
            )
                
```

It doesn't seem like we're testing this functionality currently.

Reply:
It doesn't look like that functionality has ever been added at all tbh, or at least not for signals, which would get caught by this if in any case and not initialised in any case...
```if (
            (is_device_vector and (not device_vectors or name not in device_vectors))
            or ((origin := get_origin(device_type)) and issubclass(origin, Signal))
            or (isclass(device_type) and issubclass(device_type, Signal))
        ):
```
I thought from looking at the code that signals should not pre-initialised anyway? Or did I get that wrong?

Reply:
> I thought from looking at the code that signals should not pre-initialised anyway? Or did I get that wrong?

Ah yes, my mistake. It would be
``` python
                included_optional_fields=("optional_device_with_a_signal",),
```